src/Microsoft.DotNet.CodeFormatter.Analyzers/ProvideExplicitVariableTypeFixer.cs

```diff
@@ -84,7 +84,15 @@ public override FixAllProvider GetFixAllProvider()
                                           .WithAdditionalAnnotations(Simplifier.Annotation)
                                           .WithTriviaFrom(varNode);
             documentEditor.ReplaceNode(varNode, explicitTypeNode);
-            return documentEditor.GetChangedDocument();
+            var newDocument = documentEditor.GetChangedDocument();
+
+            // We don't want the explicit type to be fully qualified. So add an using for this node and the simplifier will
+            // take care of removing it if it isn't necessary.
+            // The second parmaeter to AddImportsAsync is an annotation that is used to locate the span
+            // for which an using should be added. Since we added the simplify annotation on explicitTypeNode, 
+            // we can just use that annotation to locate the node again in newDocument. 
```

Comment:
In the current implementation, the simplifier is called immediately (during PostProcess) on a codefix and the given document shouldn't have any simplifier annotations. Having said that I was being lazy here and depending on an implementation detail. Fixed it to create a separate annotation instead.